### PR TITLE
[Update]: configuration for publishing npm package

### PR DESCRIPTION
**Pull request by Mohsen-T**
- Updated the package.json for publishing npm package
- Added .npmignore
- Added 'hardhat-abi-exporter' package for generating ABIs separately
- Ticket: [Publish NPM package with contracts](https://blox-ensemble.monday.com/boards/4802651069/pulses/4802659002)

### CODE REVIEW

File: hardhat.config.ts
Comment (mtabasco) [inline hardhat.config.ts:78]: Users will only interact with our main contracts SSVNetwork and SSVNetworkViews. So only these ABIs are needed. I think using the `only` property here you can got it.

**Review by mtabasco — CHANGES_REQUESTED** [review]
Please add a reference for the publishing process in the docs.
Also, we need to setup the NPM_TOKEN in github. Assign it to me on the Monday task as a check to have it ready before merging this PR.
Is there a way of testing this publish process or we must wait until the next release? thanks!

**Review by andrew-blox — COMMENTED** [review]
I see marco left a comment like this maybe its the same.  Why do we publish all the ABI's is it needed?  Can we do just ABI for network and views?

Just double checking this will be public package not private right?

File: docs/publish.md
Comment (mtabasco) [inline docs/publish.md:72]: Note that the publish action will occur when generating a release, not pushing to main.

File: docs/publish.md
Comment (mtabasco) [inline docs/publish.md:38]: @Mohsen-T I think instead of putting putting here the publish script, just reference to our `.github/workflows/publish.yaml` as example.

**Review by mtabasco — CHANGES_REQUESTED** [review]

**Comment by mtabasco** [comment]
> I see marco left a comment like this maybe its the same. Why do we publish all the ABI's is it needed? Can we do just ABI for network and views?
> 
> Just double checking this will be public package not private right?

@andrew-blox correct, public package

**Review by mtabasco — CHANGES_REQUESTED** [review]
Minor changes

**Comment by Mohsen-T** [comment]
> Please add a reference for the publishing process in the docs. Also, we need to setup the NPM_TOKEN in github. Assign it to me on the Monday task as a check to have it ready before merging this PR. Is there a way of testing this publish process or we must wait until the next release? thanks!

Introduced publish.md to the documentation and made updates to other Markdown files within the docs directory, enhancing the project's documentation quality.

File: docs/publish.md
Comment (Mohsen-T) [reply docs/publish.md:38]: okay

**Review by mtabasco — APPROVED** [review]

**Review by andrew-blox — APPROVED** [review]

**Pull request merged**
